runelite-client/src/main/java/net/runelite/client/util/WorldMapView.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (c) 2018 Morgan Lewis <https://github.com/MESLewis>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.util;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import java.awt.Graphics2D;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.RenderingHints;
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import javax.imageio.ImageIO;
+import javax.inject.Inject;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.coords.WorldPoint;
+
+@Slf4j
+public class WorldMapView
+{
+	private static LoadingCache<Integer, BufferedImage> regions;
+
+	private static final int REGION_GRAPHIC_SIZE = 256;
+	private static final int REGION_TILE_SIZE = 64;
+
+	@Getter
+	private WorldPoint center;
+
+	@Getter
+	private float zoom;
+
+	@Getter
+	private int graphicsWidth;
+
+	@Getter
+	private int graphicsHeight;
+
+
+	public void setCenter(WorldPoint center)
+	{
+		this.center = center;
+		updateCalculations();
```

Comment:
probably instead of doing this each time someone changes the value, something like `update` method would be better performance wise, so the operations can be done and then the update calculation can be done only once

Reply:
I do not want the user to have to call an extra method before the calculations are correct. I think updating automatically is much more user friendly than having an update method that you have to remember to call.

Reply:
I'll change it so the functions that use the pre-calculated values are responsible for updating the calculations. That is where they were originally updated until I was needing them in multiple functions so I split out that calculate method.